[WIP] Convert the GCC 2015 talk to Markdown slides using Remark

remimarenco opened the pull request:
This PR converts the GCC 2015 talk to Markdown slides using Remark (see galaxyproject/planemo@069e7ba), and refers to the issue #2244.

The end-goal is to make sure that Galaxy has great architecture documentation and a set of slides that lives with the code before [our workshop at the GCC 2016](https://gcc16.sched.org/event/5VsE/galaxy-architecture).

You can find the slides [here](https://speakerdeck.com/jxtx/galaxy-architecture-2015).
You can read more about [the remark project](https://github.com/gnab/remark) to see how to help or learn in this PR 

TODO:
- [x] Add the colors => Use style.css
- [ ] Get all the images from the slides
- [ ] Better management of the background images (See how to use [bakground-*](https://github.com/gnab/remark/wiki/Markdown#background-image) properly => Asked to reopen an issue on [remark repo](https://github.com/gnab/remark/issues/211))
- [ ] Better management of the images
- [ ] Add logo and twitter account / hashtag on first slide

remimarenco committed:
Init of #2244, second part: Python converter md to html + wip on GCC 2015 talk conversion

remimarenco committed:
Add the 2. Galaxy app architecture. Need images

remimarenco committed:
Add the 3. Galaxy components and object model

remimarenco committed:
Add more content in md

remimarenco committed:
Fix linting error by adding newline at the end of build_slideshow.py

nsoranzo commented on doc/source/slideshow/galaxy_architecture_2015.md:
```diff
@@ -0,0 +1,482 @@
+
+
+
+---
```
Empty slides can be removed

remimarenco replied:
Hi Nicola!

Thanks for the feedback, I added them because the GCC2015 presentation had 85 slides, so I anticipated them :).

remimarenco committed:
Moved the galaxyArchitecture.md into a folder dedicated to it

remimarenco committed:
Add 7th and 17th image + add style.css to have a better management of layout + added first slide as the first in GCC2015 slide

remimarenco committed:
Set properly the CSS used in the whole presentation + add special to p and first slide

remimarenco committed:
Add special orange for regular text and green emphasis (emphasis_2_galaxy)

remimarenco committed:
Changed the font to PT Sans

remimarenco committed:
Add 3. with images + few modifications

remimarenco committed:
Removed the images integrated using background-image by now Markdown imaging (still need work, image's size seems not good by default

jmchilton committed:
Add Makefile target to build architecture slides...

jmchilton committed:
2016 Updates to 2015 architecture talk.

remimarenco committed:
Merge pull request #2 from jmchilton/gcc2015_talk

Small Updates

jmchilton committed:
More intro slides for Galaxy architecture.

jmchilton committed:
Update author line.

jmchilton committed:
Progress on architecture slides...

 - Improved styling (use Oxygen - closer to Galaxy font)
 - Add ability to generate SVG diagrams to slide infrastructure
 - Add newer client-server sequence diagram based on that.

jmchilton committed:
Merge remote-tracking branch 'remimarenco/gcc2015_talk' into gcc2015_talk

jmchilton committed:
Improved logging when there are problems with SQL migrations.

jmchilton committed:
More misc work on architecture talk.

More images, slides, slide formatting, and initial work on an IPython notebook.

jmchilton committed:
Improved styling for architecture slides...

jmchilton committed:
Updates to architecture talk...

- Infrastructure for building class diagrams with plantuml
- Bring in move SVGs from last year's google docs.
- Fill out section on client architecture.

jmchilton committed:
More work on architecture talk...

jmchilton committed:
More work on slides for architecture talk.

jmchilton committed:
Add missing files from previous commit.

jmchilton committed:
Add startup_output.

jmchilton committed:
More progress on architecture slides.

jmchilton committed:
More work on architecture slides.

jmchilton committed:
Update style.

jmchilton committed:
Architecture tweaks...

jmchilton committed:
More work on the architecture talk.

bgruening commented:
@remimarenco out of interest why remark.js and not reveal-js? We are searching for a standard for the training material repository: https://github.com/bgruening/training-material/issues/8

jmchilton commented:
@remimarenco I worry about merging this and adding a bunch of large images into the git history that we don't use. May I rebase this and filter the images out of your contributions?

jmchilton committed:
Merge remote-tracking branch 'jmchilton/dev' into gcc2015_talk

Conflicts:
	Makefile

jmchilton committed:
Remove unused png images from architecture stuff.

jmchilton committed:
Remove unused architecture slides added by me.

jmchilton committed:
Replace mermaid diagrams with plantuml diagrams.

Fill out these diagrams more and style them for a black background in Galaxy oranges.

jmchilton committed:
The second half of that last commit.

Forgot to git add it all.

jmchilton committed:
Update Makefile target for slides.

Rework for consistency with rest of Makefile and add an export target.

jmchilton committed:
Arch-slides - eliminate use of ``background-image`` so we can just drop this in the docs.

jmchilton committed:
Move architecture docs to doc/source/slideshow/architecture/galaxy_architecture.md

From doc/source/slideshow/galaxy_architecture_2015/galaxy_architecture_2015.md.

remimarenco commented:
@bgruening Sorry for the delay, I missed this notification. I followed @jmchilton lead, and I think we wanted to try remark.js :).

@jmchilton Sure! Thank you!

martenson commented:
I rebased this into the PR linked above

martenson commented:
both @remimarenco and @jmchilton agreed with the rebase, closing
